### PR TITLE
Images part 2

### PR DESCRIPTION
**Pull request by ghost**
Fixed what was a failed attempt at images part 2 XD! 
Tested beforehand and just pulled from master so should be good!

### CODE REVIEW

**Comment by huang27c** [comment]
Looks good to me!

**Review by nnunley — APPROVED** [review]

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -88,8 +97,13 @@ public void storeMessage(Message message) {
   /**
    * Gets messages posted by a specific user.
    *
+   * <p><<<<<<< HEAD
+   *
+   * @return a list of messages posted by the user, or empty list if user has never posted a
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Datastore.java]: Looks like a bad merge here

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -88,8 +97,13 @@ public void storeMessage(Message message) {
...
+   *     message. List is sorted by time descending. =======
    * @return a list of messages posted by the user, or empty list if user has never posted a
-   *     message. List is sorted by time descending.
+   *     message. List is sorted by time descending. >>>>>>>
+   *     037bfe5090a14815ff884f5734713fbdf018df1b
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Datastore.java]: And here

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 3]
```diff
@@ -62,34 +68,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<Message> messages = datastore.getMessages(user);
     Gson gson = new Gson();
     String json = gson.toJson(messages);
+    System.out.println(json);
```
Comment (nnunley) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: Remove or replace with java logging.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 4]
```diff
@@ -62,34 +68,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+          "(?:^|[\\W])((ht|f)tp(s?):\\/\\/|www\\.)"
+              + "(([\\w\\-]+\\.){1,}?([\\w\\-.~]+\\/?)*"
+              + "[\\p{Alnum}.,%_=?&#\\-+()\\[\\]\\*$~@!:/{};']*)",
+          Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL);
 
   /** Stores a new {@link Message}. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
+    System.out.println("doPost!!!!!!!!!!!!!!!");
```
Comment (nnunley) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: And again

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 5]
```diff
@@ -62,34 +68,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
     UserService userService = UserServiceFactory.getUserService();
     if (!userService.isUserLoggedIn()) {
       response.sendRedirect("/index.html");
       return;
     }
+    String requestData = request.getReader().lines().collect(Collectors.joining());
```
Comment (nnunley) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: And here.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 6]
```diff
@@ -62,34 +68,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
     String replacement = "<img src=\"$1\" />";
     String textWithImagesReplaced = userText.replaceAll(regex, replacement);
 
-    Message message = new Message(user, textWithImagesReplaced, recipient);
+    Message message = new Message(user, textWithImagesReplaced, recipient, "");
+
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+    System.out.println("blobkeys" + blobKeys);
```
Comment (nnunley) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: And here.

File: src/main/webapp/user-page.html [hunk 7]
```diff
@@ -68,17 +68,19 @@ <h1 id="page-title">User Page</h1>
...
+    
+  
+    <form id="message-form" action="/messages" method="POST" class="hidden" enctype="multipart/form-data">
+      Enter a new message:!
+      <br/> 
+      <textarea name="text" id="message-input"> </textarea>
       <br/>
-      Add an image to your message:
+      Add an image to your message:!
       <input type="file" name="image" >
```
Comment (nnunley) [inline src/main/webapp/user-page.html]: Did you mean to add the ! at the end?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 4 above)
Comment (nnunley) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: And more debugging stuff that should either be a logging line or removed.

File: src/main/java/com/google/codeu/data/Message.java [hunk 8]
```diff
@@ -84,9 +92,26 @@ public long getTimestamp() {
...
+   *
    * @return recipient
    */
   public String getRecipient() {
     return recipient;
   }
+
+  /** Sets current imageUrl. */
+  public void setImageUrl(String imageUrl) {
+    System.out.println("imageurl");
```
Comment (nnunley) [inline src/main/java/com/google/codeu/data/Message.java]: And more logging stuff.

**Review by nnunley — CHANGES_REQUESTED** [review]

**Review by nnunley — APPROVED** [review]

**Pull request merged**